src/main/java/com/mongodb/hibernate/dialect/MongoDialect.java

```diff
@@ -46,6 +46,6 @@ protected DatabaseVersion getMinimumSupportedVersion() {
 
     @Override
     public SqlAstTranslatorFactory getSqlAstTranslatorFactory() {
-        return new MongoTranslatorFactory();
+        return MongoTranslatorFactory.INSTANCE;
```

Comment:
This way we work around the issue fixed in https://github.com/hibernate/hibernate-orm/pull/9900, because we are not getting that fix in Hibernate ORM 6.x.

Reply:
Thanks for your raising the issue and it is a quick fix. On the other hand, it seems trivial to me for we only avoid one unnecessary duplicated object creation (which usually is not a big deal for the creation is not costly).

Reply:
The other id generator defect is more important and hard to fix from Hibernate side. I created a PR at https://github.com/hibernate/hibernate-orm/pull/9965. Let us see.